GVFS/GVFS.UnitTests/Git/GitAuthenticationTests.cs

```diff
@@ -154,36 +155,89 @@ public void TwoThreadsInterleavingFailuresShouldntStompASuccess()
             dut.TryGetCredentials(tracer, out thread2Auth, out error).ShouldEqual(true);
 
             // Simulate a 401 error on one threads
-            dut.Revoke(thread1Auth);
+            dut.RejectCredentials(thread1Auth);
 
             // That thread then retries and succeeds
             dut.TryGetCredentials(tracer, out thread1Auth, out error).ShouldEqual(true);
-            dut.ConfirmCredentialsWorked(thread1Auth);
+            dut.ApproveCredentials(thread1Auth);
 
             // If the second thread fails with the old PAT, it shouldn't stomp the new PAT
-            dut.Revoke(thread2Auth);
+            dut.RejectCredentials(thread2Auth);
 
             // The second thread should be able to get a PAT
             dut.TryGetCredentials(tracer, out thread2Auth, out error).ShouldEqual(true);
             thread2Auth.ShouldEqual(thread1Auth, "The second thread stomp the first threads good auth string");
         }
 
+        [TestCase]
+        public void DontDoubleStoreExistingCredential()
+        {
+            MockTracer tracer = new MockTracer();
+            MockGitProcess gitProcess = this.GetGitProcess();
+
+            GitAuthentication dut = new GitAuthentication(gitProcess, "mock://repoUrl");

```

Comment:
Nit: since you use "mock://repoUrl" in several places you could assign it to a string